datastore/postgres/matcher_store.go

```diff
@@ -83,3 +86,102 @@ func (s *MatcherStore) RecordUpdaterStatus(ctx context.Context, updaterName stri
 func (s *MatcherStore) RecordUpdaterSetStatus(ctx context.Context, updaterSet string, updateTime time.Time) error {
 	return recordUpdaterSetStatus(ctx, s.pool, updaterSet, updateTime)
 }
+
+func (s *MatcherStore) GetLatestVulnerabilities(ctx context.Context, updater string) ([]*claircore.Vulnerability, error) {
```

Comment:
Does this need to load all the data eagerly? An iterator+map façade might get us a reduced memory footprint?

Reply:
That'd probably be an improvement :thinking: I'd have to think about a map that lazily fetches it's values though... I think we really need to nail-down what these DeltaUpdaters are going to do with the information. I tried to capture some of that in the test `testUpdater.DeltaParse()`.

Reply:
I think the choice of using a client-calculated digest is going to bite us here, as we would need to expose it in order to do an equality check.

Reply:
I have updated the PR and feel a lot better about the flow now. The updater just needs to signal that the data it's fetching are deltas and the manager will choose the `DeltaUpdatevulnerabilities()`. All the updating is captured in this function. It's job is to:
* Get existing vulnerabilities (just ID and name will suffice)
* Delete any existing vulnerabilities that appear in the new batch
* Create the uo (as is tradition)
* Add the existing vulns to the new uo
* Persist the new vulns to the DB (as is tradition)